app/services/search/vacancy_sort.rb

```diff
@@ -1,16 +1,35 @@
 class Search::VacancySort < RecordSort
-  def initialize(keyword:)
+  def initialize(keyword:, location: nil)
     @keyword = keyword
+    @location = location
     super
   end
 
   def options
-    [publish_on_desc_option, closing_date_asc_option]
+    if location.present?
+      [distance_option, publish_on_non_default_desc_option, closing_date_asc_option]
+    else
+      [publish_on_desc_option, closing_date_asc_option]
+    end
+  end
+
+  def default_sort_option
+    if location.present?
+      distance_option
+    else
+      publish_on_desc_option
+    end
+  end
+
+  def by_db_column?
+    return true if sort_by == "publish_on_non_default"
```

Comment:
this is to allow "publish_on_non_default" to functionally behave like "publish_on"